### PR TITLE
floatingpoint: Use unique_ptr for FloatingPointLiteral pointer.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/util/floatingpoint.cpp [hunk 1]
```diff
@@ -129,19 +129,19 @@ FloatingPoint::FloatingPoint(const FloatingPointSize& size,
...
-    d_fpl = new FloatingPointLiteral(
+    d_fpl.reset(new FloatingPointLiteral(
         symfpu::convertUBVToFloat<symfpuLiteral::traits>(
             symfpuLiteral::CVC4FPSize(size),
             symfpuLiteral::CVC4RM(rm),
-            symfpuLiteral::CVC4UnsignedBitVector(bv)));
+            symfpuLiteral::CVC4UnsignedBitVector(bv))));
   }
 #else
   d_fpl = new FloatingPointLiteral(2, 2, 0.0);
```
Comment (nafur) [inline src/util/floatingpoint.cpp:147]:
```suggestion
  d_fpl.reset(new FloatingPointLiteral(2, 2, 0.0));
```

**Review by nafur — COMMENTED** [review]
To fix compilation error...

**Review by 4tXJ7f — COMMENTED** [review]
I think that it would be a good idea to also fix the issue that I've outlined here: https://github.com/CVC4/CVC4/commit/fedb3256b37511943c4a843327d36da31480be69#r44639752 in this PR since it is quite related. (Not marking my comment as "request changes" though in case you prefer to submit a separate PR for that)

**Comment by aniemetz** [comment]
@4tXJ7f https://github.com/CVC4/CVC4/commit/fedb3256b37511943c4a843327d36da31480be69#r44669395

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the update.

**Pull request merged**